docs/cpp/Runner.md

```diff
@@ -0,0 +1,221 @@
+          
+      Saves results of computations stored in a structure of the TOutput type to the output files:
+      
+        - output_prefix + input_basename + "\_F\_" + module_basename + ".txt" - stores the value of the objective function
+        - output_prefix + input_basename + "\_J\_" + module_basename + ".txt" - stores the value of the objective function derivative
+          
+      The format of the output files is specific for each objective type.
+4.  Include "TBenchmark.h" in "main.cpp" and add a new else-if branch to the "main" function as follows:
+    ```
+    if (test_type == "GMM") {
+        run_benchmark<GMMInput, GMMOutput, GMMParameters>(module_path, input_filepath, output_prefix, minimum_measurable_time, nruns_F, nruns_J, time_limit, { replicate_point });
+    }
+        ...
+    else if (test_type == "T") {
+            run_benchmark<TInput, TOutput, TParameters>(module_path, input_filepath, output_prefix, minimum_measurable_time, nruns_F, nruns_J, time_limit, parameters);
+    }  
+        ...
+    ```
+5. Add *TBenchmark.cpp* to *TestExtenders* environment variable inside "/src/cpp/runner/CMakeLists.txt". 
+
+## Input/Output files format
+
+### GMM
+#### Input
+  D k n</br>
+  α<sub>1</sub></br>
+    ...</br>
+  α<sub>k</sub></br>
+  μ<sub>1,1</sub> ... μ<sub>D,1</sub></br>
+    ...</br>
+  μ<sub>1,k</sub> ... μ<sub>D,k</sub></br>
+  q<sub>1,1</sub> ... q<sub>D,1</sub> l<sub>1,1</sub> ... l<sub>$\frac{D(D-1)}{2}$,1</sub></br>
+    ...</br>
+  q<sub>1,k</sub> ... q<sub>D,k</sub> l<sub>1,k</sub> ... l<sub>$\frac{D(D-1)}{2}$,k</sub></br> 
+  x<sub>1,1</sub> ... x<sub>D,1</sub></br>
+    ...</br>
+  x<sub>1,n</sub> ... x<sub>D,n</sub></br>
+  γ m</br>
+  
+Definitions of all variables are given in the  [srajer-autodiff-screen.pdf](../../Documents/srajer-autodiff-screen.pdf), page 3.
+Note that if replicate point mode is enabled the benchmark expects only x<sub>1,1</sub> ... x<sub>D,1</sub> string and duplicates it n-1 times.
+
+#### Output
+
+1. \_F\_ file  
+    Contains only the value of the function in the specified point. 
+2. \_J\_ file  
+     v<sub>1</sub> ... v<sub>n</sub>      where v<sub>i</sub> are components of the objective gradient.
+     
+### BA
+#### Input
+  n m p</br>
+  p<sub>1</sub> ... p<sub>11</sub></br>
+  x<sub>1</sub> x<sub>2</sub> x<sub>3</sub></br>
+  w<sub>1</sub></br>
+  feat<sub>1</sub> feat<sub>2</sub></br>
+
+n,m,p are number of cams, points and observations.
+Definitions of all other variables are given in the  [srajer-autodiff-screen.pdf](../../Documents/srajer-autodiff-screen.pdf), page 4.
+
+#### Output
+
+1. \_F\_ file  
+    Reprojection error
+    Zach weight error
+2. \_J\_ file  
+     j<sub>1,1</sub> ... j<sub>1,ncols</sub> 
+     ...
+      j<sub>nrows,1</sub> ... j<sub>nrows,ncols</sub> 
+    
+    where ncols=2*p+p, nrows=11*n+3*m+p
+
+### Hand
+#### Input
+1. model/bones.txt
+	Contains a list of lines whereas each line containts such parameters separated by ":" delimeter:
```

Comment:
'whereas' means 'while'. Probably, you meant just 'where. Same below.
Also,
...each line _contain~t~s_ _the following_ parameters...